WebService/v1/Models/SimulationApiModel/SimulationRateLimits.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Concurrency;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Models.SimulationApiModel
+{
+    public class SimulationRateLimits
+    {
+        [JsonProperty(PropertyName = "ConnectionsPerSecond", NullValueHandling = NullValueHandling.Ignore)]
+        public int ConnectionsPerSecond { get; set; }
+
+        [JsonProperty(PropertyName = "RegistryOperationsPerMinute", NullValueHandling = NullValueHandling.Ignore)]
+        public int RegistryOperationsPerMinute { get; set; }
+
+        [JsonProperty(PropertyName = "TwinReadsPerSecond", NullValueHandling = NullValueHandling.Ignore)]
+        public int TwinReadsPerSecond { get; set; }
+
+        [JsonProperty(PropertyName = "TwinWritesPerSecond", NullValueHandling = NullValueHandling.Ignore)]
+        public int TwinWritesPerSecond { get; set; }
+
+        [JsonProperty(PropertyName = "DeviceMessagesPerSecond", NullValueHandling = NullValueHandling.Ignore)]
+        public int DeviceMessagesPerSecond { get; set; }
+
+        // Default constructor used by web service requests
+        public SimulationRateLimits()
+        {
+            this.ConnectionsPerSecond = 0;
+            this.RegistryOperationsPerMinute = 0;
+            this.TwinReadsPerSecond = 0;
+            this.TwinWritesPerSecond = 0;
+            this.DeviceMessagesPerSecond = 0;
+        }
+
+        // Map API model to service model
+        public static SimulationRateLimits FromServiceModel(Services.Models.Simulation.SimulationRateLimits simulationRateLimits, IRateLimiting rateReporter)
+        {
+            var defaultRatingLimits = rateReporter.GetCounters();
+
+            var connectionsPerSecond = simulationRateLimits?.ConnectionsPerSecond > 0 ? simulationRateLimits.ConnectionsPerSecond : defaultRatingLimits.ConnectionsPerSecond;
```

Comment:
What happens if simulationRateLimits is null? What does connectionsPerSecond get set to? Perhaps we should have a check at the top of the method for the input not being null?

Reply:
If simulationRateLimits  is null, then we will fall back to defaultRatingLimits. I have tested this scenario